src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,34 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_TIME_FORMATTER =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder resultStr = new StringBuilder();
+        resultStr.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        for (String name : names) {
+            resultStr.append(System.lineSeparator()).append(name).append(" - ");
```

Comment:
I can see two constants here:
```
private static final String REPORT_START = "Report for period ";
private static final String HYPHEN = " - ";
```

```suggestion
        resultStr.append(REPORT_START).append(dateFrom).append(HYPHEN).append(dateTo);
        for (String name : names) {
            resultStr.append(System.lineSeparator()).append(name).append(HYPHEN);
```

Reply:
you haven't declared the necessary constants :)